pkg/test/server.go

```diff
@@ -66,6 +66,7 @@ func RunHTTP(ctx context.Context, upstreamPort uint) {
 			log.Error(err)
 		}
 	}()
+	<-ctx.Done()
 	if err := server.Shutdown(ctx); err != nil {
```

Comment:
isn't this using an expired context now? will the clean shutdown actually do anything?

Reply:
I'm admittedly very inexperienced with go, so I may be wrong, but from reading the documentation on Server.Shutdown (https://golang.org/pkg/net/http/?#Server.Shutdown), I think you are correct. My understanding is that it is intended for when you're about to shut down a server, so that you finish processing any outstanding work, without accepting new connections.

I actually can't think of why it was being called in the first place in this test. We know exactly how much work to do, so there should be no surprises awaiting clients when we exit at the end of the test. Do you have any idea why?

As for a better fix, do I even need the <-ctx.Done(), or should I just remove it and the shutdown all-together? The test seems to pass without it, but I guess this could be used as a way to keep the outer goroutine from exiting prior to the test being finished? Is that good practice?

Reply:
I guess the way I would do it is create a sub context with `context.WithCancel`, so that the context can be cancelled, which will actually end the sub context and continue to the server shutdown with the parent context.  I'm not sure how you would ensure the cancel function is called when the tests finish.  Just leaving out the clean shutdown might be fine too.

Another option might be to switch to using https://golang.org/pkg/net/http/httptest/ since this is a test package.

Reply:
I just removed it all-together.